tstrait/simulate_phenotype.py

```diff
@@ -250,7 +253,10 @@ def sim_genetic_value(self):
                 2 * len(individual_genotype)
             )
             beta_array[i] = self.model.sim_effect_size(
```

Comment:
This looks good @daikitag , but I think we can simplify a bit by entirely decoupling the frequency dependence for beta from the trait model like so:
```python
beta = self.model.sim_effect_size(self.num_causal, self.rng)
beta_array[i] = frequency_dependence(allele_frequency[i], self.alpha)
```

So the trait distribution really is just drawing ``num_causal`` values from a given numpy random distribution.
```

Reply:
@jeromekelleher 
Thank you very much for your feedback. I modified the codes to add the frequency dependence in the simulation part, and I also restructured the unit tests and statistical tests.